### PR TITLE
Allow to specify build platform for test images

### PR DESCRIPTION
**Pull request by BbolroC**
## Description

This is an echo of the https://github.com/knative/eventing/pull/4763 for client. The test images are built and uploaded without a break on other architectures by specifying a `PLATFORM` env variable.

## Changes

* Add a description to test/README.md
* Add a local variable `platform` (empty by default) which can be set by an environment variable `PLATFORM` at test/upload-test-images.sh

### CODE REVIEW

**Comment by dsimansk** [comment]
Thanks!

/ok-to-test

**Review by rhuss — APPROVED** [review]
/lgtm
/approve

File: test/README.md [hunk 1]
```diff
@@ -85,6 +85,10 @@ To run the script for all end to end test images:
 A docker tag may be passed as an optional parameter. This can be useful on
 Minikube in tandem with the `--tag` [flag](#using-a-docker-tag):
 
+`PLATFORM` environment variable is optional. If it is specified, test images
```
Comment (maximilien) [inline test/README.md:88]: Should this be `PLATFORMS` with s since the point is there could be more than one?

**Review by maximilien — CHANGES_REQUESTED** [review]
See if my one change comment makes sense.

**Pull request merged**